code/game/objects/items/busterarm.dm

```diff
@@ -0,0 +1,913 @@
+/obj/effect/proc_holder/spell/targeted/buster
+	clothes_req = FALSE
+	include_user = TRUE
+	range = -1
+
+/obj/effect/proc_holder/spell/targeted/buster/can_cast(mob/living/user)
+	var/obj/item/bodypart/l_arm/robot/buster/L = user.get_bodypart(BODY_ZONE_L_ARM)
+	var/obj/item/bodypart/r_arm/robot/buster/R = user.get_bodypart(BODY_ZONE_R_ARM)
+	if(R?.bodypart_disabled || L?.bodypart_disabled)
+		to_chat(user, span_warning("The arm isn't in a functional state right now!"))
+		return FALSE
+	if(user.IsParalyzed() || user.IsStun() || user.restrained())
+		return FALSE
+	return TRUE
+	
+/obj/item/buster/proc/hit(mob/living/user, mob/living/target, damage)
+		var/obj/item/bodypart/limb_to_hit = target.get_bodypart(user.zone_selected)
+		var/armor = target.run_armor_check(limb_to_hit, MELEE, armour_penetration = 35)
+		target.apply_damage(damage, BRUTE, limb_to_hit, armor, wound_bonus=CANT_WOUND)
+
+/obj/effect/proc_holder/spell/targeted/buster/proc/grab(mob/living/user, mob/living/target, damage)
+		var/obj/item/bodypart/limb_to_hit = target.get_bodypart(user.zone_selected)
+		var/armor = target.run_armor_check(limb_to_hit, MELEE, armour_penetration = 35)
+		target.apply_damage(damage, BRUTE, limb_to_hit, armor, wound_bonus=CANT_WOUND)
+
+/obj/item/buster
+	item_flags = DROPDEL
+	w_class = 5
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	lefthand_file = 'icons/mob/inhands/misc/touchspell_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/touchspell_righthand.dmi'
+
+/obj/item/buster/Initialize(mob/living/user)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, HAND_REPLACEMENT_TRAIT)
+
+/obj/effect/proc_holder/spell/targeted/buster/wire_snatch
+	name = "Wire Snatch"
+	desc = "Extend a wire to your active hand for reeling in foes from a distance. Reeled in targets will be unable to walk for 1.5 seconds. Anchored targets that are hit will\
+	pull you towards them instead. It can be used 3 times before reeling back into the arm."
+	invocation_type = "none"
+	include_user = TRUE
+	range = -1
+	charge_max = 50
+	clothes_req = FALSE
+	cooldown_min = 10
+	action_icon = 'icons/obj/guns/magic.dmi'
+	action_icon_state = "hook"
+	var/summon_path = /obj/item/gun/magic/wire
+
+/obj/effect/proc_holder/spell/targeted/buster/wire_snatch/cast(list/targets, mob/user)
+	for(var/obj/item/gun/magic/wire/T in user)
+		qdel(T)
+		to_chat(user, span_notice("The wire returns into your wrist."))
+		return
+	for(var/mob/living/carbon/C in targets)
+		var/GUN = new summon_path
+		C.put_in_r_hand(GUN)
```

Comment:
if the arm can be on either side of the body, why put the wire specifically in the right hand?

Reply:
the default version of the spell is for the left hand while the subtype is for the right